### PR TITLE
Adding an ansible affinity group example to the ansible project.

### PR DESCRIPTION
**Pull request by stvnrsnbrg**
Adding an ansible affinity group example to the ansible project.

### CODE REVIEW

**Comment by ovirt-infra** [comment]
Hello contributor, thanks for submitting a PR for this project! 

I am the bot who triggers "standard-CI" builds for this project.
As a security measure, I will not run automated tests on PRs that are not from white-listed contributors.

In order to allow automated tests to run, please ask one of the project maintainers to review the code and then do one of the following:

1. Type `ci test please` on this PR to trigger automated tests for it.
2. Type `ci add to whitelist` on this PR to trigger automated tests for it and also add you to the contributor white-list so that your future PRs will be tested automatically.
3. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the project organization, so you'll be able to run tests for all the organization's projects.

**Comment by mykaul** [comment]
Nice. I wonder how common it is to provide documentation within the playbook.
Perhaps the vars should be in a separate file?

**Comment by machacekondra** [comment]
@mykaul I think it should be in `examples` directory as it currently is. We ship this directory with RPM in `/usr/share/doc/{packagename}/examples`. It will be shipped also from galaxy as whole git repo is installed.

**Comment by machacekondra** [comment]
@MarSik

**Comment by MarSik** [comment]
It should get some documentation at the top even in examples. Just to make sure users understand what the example is doing.

Btw, I believe a single strong negative affinity group A-B should be enough to achieve what Yaniv wanted as it actually does not matter where the two VMs are running as long as they are not on the same host.

**Comment by MarSik** [comment]
And one other note.. Ondra should probably tell us whether the role creates the VMs and adds the hosts too or whether it uses the vars just as references.

**Comment by machacekondra** [comment]
It doesn't create the hosts, it only creates a VMs.

**Comment by MarSik** [comment]
I am not sure that is desirable in this case.. it might be, but the example must then mention that it creates both VMs and the affinity that makes them suitable for HA load

**Comment by mykaul** [comment]
@machacekondra - I actually did not want this as an example, but as a reusable role others can use.
For example, OpenShift installation (https://github.com/openshift/openshift-ansible-contrib/blob/master/reference-architecture/rhv-ansible/playbooks/vars/ovirt-infra-vars.yaml#L116 ) might be able to use this on a hyper-converged specific infra., etc.

**Comment by MarSik** [comment]
@mykaul Is there a reason you wanted to limit where the VMs are allowed to run instead of just making sure they never run together?

**Comment by mykaul** [comment]
@MarSik - no reason to limit - just a simple use case - RHHI (hyper-converged). I feel a simple role that can be extended is better than an abstract one.

**Comment by machacekondra** [comment]
@mykaul Well, that role you want is this role, actually. So is this missing some features? I think I still don't understand what is needed here.

**Comment by mykaul** [comment]
How can people re-use it? Perhaps I wasn't clear - I do not want people to
re-write this piece of code over and over, but re-use this simple playbook
- perhaps it should be a task?
Given 2 VMs and 3 hosts, create anti-affinity between the VMs and
soft-affinity between the hosts and the VMs properly (as was done in this
PR).

I think it's useful to any given HA VMs pair.

**Comment by machacekondra** [comment]
In that case having such example is the best what we can have I think. I will review and merge.

File: examples/ovirt_affinity_infra.yml [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  hosts: localhost
+  connection: local
+  gather_facts: false
+
+  vars_files:
+    # Contains encrypted `engine_password` varibale using ansible-vault
+    - passwords.yml
+    
+  vars:
+    engine_url: http://localhost:8080/ovirt-engine/api
```
Comment (machacekondra) [inline examples/ovirt_affinity_infra.yml:12]: We use this in all examples:
```
engine_url: //ovirt-engine.example.com/ovirt-engine/api
```

File: examples/ovirt_affinity_infra.yml [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    engine_user: admin@internal
+    # engine_cafile: /etc/pki/ovirt-engine/ca.pem
```
Comment (machacekondra) [inline examples/ovirt_affinity_infra.yml:14]: Please uncomment.

File: examples/ovirt_affinity_infra.yml [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+    vms:
+      - name: VM1
+        tag: dist_vm
+        profile: "{{ dist_vm }}"
+      - name: VM2
+        tag: dist_vm
+        profile: "{{ dist_vm }}"
+
+    hosts:
```
Comment (machacekondra) [inline examples/ovirt_affinity_infra.yml:30]: Please remove, we don't add any hosts by this role.

File: examples/ovirt_affinity_infra.yml [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    
+  vars:
+    engine_url: http://localhost:8080/ovirt-engine/api
+    engine_user: admin@internal
+    # engine_cafile: /etc/pki/ovirt-engine/ca.pem
+    
+    dist_vm:
+      cluster: Default
+
+    phys_host:
```
Comment (machacekondra) [inline examples/ovirt_affinity_infra.yml:19]: Please remove.

File: examples/ovirt_affinity_infra.yml [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    # Contains encrypted `engine_password` varibale using ansible-vault
+    - passwords.yml
+    
+  vars:
+    engine_url: http://localhost:8080/ovirt-engine/api
+    engine_user: admin@internal
+    # engine_cafile: /etc/pki/ovirt-engine/ca.pem
+    
+    dist_vm:
+      cluster: Default
```
Comment (machacekondra) [inline examples/ovirt_affinity_infra.yml:17]: I would add also a `template` for the profile. For example:
```yaml
dist_vm:
  cluster: Default
  template: rhel74
```

**Review by machacekondra — CHANGES_REQUESTED** [review]
Looks OK in general, few comments.

**Comment by MarSik** [comment]
@mykaul But there is no reason to limit the VMs more than by anti-affinity between them. So the question is whether this is supposed to be a production best practices example or an artificial feature showcase.

**Comment by machacekondra** [comment]
Please re-open if still valid.

**Pull request closed without merging**